.github/workflows/test.yaml

```diff
@@ -20,31 +20,40 @@ jobs:
   testing:
     runs-on: ubuntu-20.04
     env:
-      TOXENV: "${{ matrix.tox-env }}-withcov-posix,coverage-prepare,codecov-push,coveralls-push"
+      TOXENV: "${{ matrix.tox-env }}"
+      CODECOV_OPTIONS: "-n 'lnx-${{ matrix.python-version }}-${{ matrix.tox-env }}${{ matrix.noipv6 }}'"
+      TRIAL_ARGS: "${{ matrix.trial-args }}"
     name: ${{ matrix.python-version }}${{ matrix.noipv6 }}-${{ matrix.tox-env }}-withcov-posix
     strategy:
+      fail-fast: false
       matrix:
         # Run on the minimum micro Python version that we can get on CI.
         # When updating the minimum Python version here, also update the
         # `python_requires` from `setup.cfg`.
         # Run on latest minor release of each major python version.
         python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-alpha.5, pypy-3.6, pypy-3.7]
-        tox-env: [alldeps]
+        tox-env: ['alldeps-withcov-posix']
         noipv6: ['']
+        # As of 9 March 2021 GHA VM have 2 CPUs - Azure Standard_DS2_v2
```

Comment:
Typically you want to go a little bit above the raw CPU count, since processes can and do also block on I/O, so you want to give the scheduler a bit more ready work to play with.  It really depends on the workload though, so my suggestion would be to bump it to 3 and look at the timings.

Reply:
I did some test on my local system before going with this value :) ... but increasing above CPU count yields no improvements... But I see that on GitHub VM there is a small improvement.

The GitHub VM are slower... so 3minutes for a run vs 20seconds for local run

Below are my local results on a 8core 16 threads cpu... i did multiple runs for each job count... here are just some examples:

* -j8 Ran 11952 tests in 27.908s
* -j16 Ran 11952 tests in 17.072s
* -j17 Ran 11952 tests in 17.965s
* -j24 Ran 11952 tests in 17.671s

---------

I went with -j2 as this is the CPU count on Azure, but at the same time, to reduce the probability of running at the same time tests that are not isolated.

Here are the logs for -j2 https://github.com/twisted/twisted/runs/2075976363
Here are the logs for -j4 https://github.com/twisted/twisted/pull/1545/checks?check_run_id=2085348540

There is an increase in performance for -j4

On -j4 pypy tests failed 

For the initial change I would say that is better to do baby steps and just go with 2 concurrent jobs to not introduce (just yet) too many random failures on trunk.

Reply:
> Below are my local results on a 8core 16 threads cpu... i did multiple runs for each job count... here are just some examples:

FWIW I would expect the top speed to be somewhere around `-j11` or `-j10` for an 8-core machine, not all the way up at `-j16`.  (In my experience hyperthreading doesn't help Python very much, since there's too much cache contention; it's real cores that give you real concurrency.)

Reply:
> For the initial change I would say that is better to do baby steps and just go with 2 concurrent jobs to not introduce (just yet) too many random failures on trunk.

Yep, that seems fine.

Reply:
With -j10 I only got 19 seconds... and j11 got 18 seconds... but yes j14 got the same results as j16 --- 17 seconds.

On my local system I have executed the full Twisted tests with python 3.9 and pypy3.6 hundreds of times without any random failure.